plugins/inputs/socket_listener/socket_listener.go

```diff
@@ -313,6 +338,14 @@ func (uc unixCloser) Close() error {
 	return err
 }
 
+func (uc unixCloser) Accept() (net.Conn, error) {
+	return uc.closer.(net.Listener).Accept()
```

Comment:
if we're assuming that `uc.closer` is always going to be a `net.Listener`, you can avoid the type assertion by redefining `uc.closer` to be a `net.Listener`.
This is good practice anyway as then you get compile time verification that you never assign anything other than a `net.Listener` to `uc.closer`, as if you do otherwise this code will panic.

Reply:
I don't see why we need these functions at all...

Reply:
if you embed `net.Listener` directly within `unixCloser` so that `unixCloser` picks up `net.Listener`'s methods, then yeah, you shouldn't.

Reply:
Oh, you mean they aren't needed because they're not even called. yeah... :-)